### PR TITLE
Shortlinks: If no wp.me shortlink, allow wp_get_shortlink() to continue

### LINKED ISSUE #8250: JetPack breaks wp_get_shortlink() instead of removing 'wp_shortlink_wp_head' hook!
JetPack has decided to return an empty string by default when `wp_get_shortlink()` is called for a custom post type. This disables the "shortlink" meta tag and "Get Shortlink" button, but ALSO breaks the WP core `wp_get_shortlink()` function! JetPack is not the only plugin out there -- `wp_get_shortlink()` is used by other plugins for a variety of reasons -- standard core functionality should not be broken on purpose. Instead of break the `wp_get_shortlink()` function, please use this instead:

```
remove_action( 'wp_head', 'wp_shortlink_wp_head' );
```

Thanks,

js.

### PR DESCRIPTION
**Pull request by jsmoriss**
Fixes #8250

#### Changes proposed in this Pull Request:

* do not change the value (if any) from previous filter hooks
* allow wp_get_shortlink() to continue and return its own shortlink
* return false, previous shortlink, or wp.me shortlink (if no previous shortlink)

#### Testing instructions:

*

### CODE REVIEW

File: modules/shortlinks.php [hunk 1]
```diff
@@ -85,5 +85,11 @@ function wpme_get_shortlink( $id = 0, $context = 'post', $allow_slugs = true ) {
 }
 
 function wpme_get_shortlink_handler( $shortlink, $id, $context, $allow_slugs ) {
-	return wpme_get_shortlink( $id, $context, $allow_slugs );
+	if ( $shortlink === false ) {	// respect / do not change the value from previous filter hooks
```
Comment (jeherve) [inline modules/shortlinks.php]: Could you use Yoda conditions here, as per WordPress' coding standards?
https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions

File: modules/shortlinks.php [hunk 2]
```diff
@@ -85,5 +85,11 @@ function wpme_get_shortlink( $id = 0, $context = 'post', $allow_slugs = true ) {
...
+		$shortlink = wpme_get_shortlink( $id, $context, $allow_slugs );	// returns wp.me shortlink or empty string
+		if ( $shortlink === '' ) {	// empty string for empty post object or no post type support for shortlinks
```
Comment (jeherve) [inline modules/shortlinks.php]: Could you use Yoda conditions here as well?

File: modules/shortlinks.php (hunk 1 above)
Comment (jsmoriss) [reply modules/shortlinks.php]: I'm not a fan, lol, but sure - done. :)

File: modules/shortlinks.php (hunk 2 above)
Comment (jsmoriss) [reply modules/shortlinks.php]: Done as well. ;-)

**Comment by kraftbj** [comment]
Looks like https://github.com/Automattic/jetpack/pull/8273/commits/6132009374aee431939de26b9993184aae9763bb picked up a lot of unintended changes upstream. Could you rebase or at least exclude that comment or whatever is needed to purify the PR to only your changes?

**Pull request closed without merging**